### ISSUE TITLE
ticklabel_format reset to default by ScalarFormatter

### ISSUE THREAD

**Issue by gepcel** [matplotlib/matplotlib#6330]
# Example:
I want to set ticklabel's as `scilimits=(-3, 3)` and `useMathText=True`.
This is what works:
```python
plt.plot(1, 10000, 2, 20000, 'ro')
ax = plt.gca()
ax.yaxis.set_major_formatter(mpl.ticker.ScalarFormatter(useMathText=True))
ax.ticklabel_format(scilimits=(-3, 3))  # put after 'set_major_formatter', scilimits works
plt.show()
```
And this is what doesn't work, by switching two line.
```python
plot(1, 10000, 2, 20000, 'ro')
ax = gca()
ax.ticklabel_format(scilimits=(-3, 3))   #put before 'set_major_formatter', scilimits doesn't work.
ax.yaxis.set_major_formatter(mpl.ticker.ScalarFormatter(useMathText=True))
plt.show()
```
# Reason:
This is what I've find out, I'd like to help, but I don't know weather this should be fixed or how this can be fixed.
In the source code [here](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/ticker.py#L458-L459), every time a `ScalarFormatter` class was called, those two values (formatter.scientific, and formatter.scilimits) was reset.

# Something to mention:
1. In matplotlib, there are always many more better ways to one same thing. If anyone know some more elegant way of doing this, please let me know. And this way of setting scilimits and useMathText is what I've learned from this very good [tutorial](https://www.youtube.com/watch?v=DNRJwENqEUY).
2. I've made a commented about this in #6276 . But I'm sure they are relevant or not, so I'm filling this issue here.

**Comment by efiring** [comment]
This behavior is fundamental; I don't think it is likely to change.  `ticklabel_format` changes properties of the existing formatter, and `yaxis.set_major_formatter` connects a new formatter to the y axis.   What would be possible is to expose more parameters such as scilimits as kwargs to the `ScalarFormatter` constructor; and/or to expose more of them (such as `useMathText`) in `ticklabel_format`.

**Comment by gepcel** [comment]
I think expose `useMathText` in ticklabel_format would be a better idea.

**Comment by gepcel** [comment]
I think to expose  `useMathText` in `ticklabel_format` would be a better idea.
If you agree with this, maybe I can make some changes.

**Comment by tacaswell** [comment]
You can use `get_major_foramatter()` and poke at it directly.

**Comment by gepcel** [comment]
Can anyone please examine this?
Thank you.

**Comment by gepcel** [comment]
Should I close this? I'm new to github.

**Comment by tacaswell** [comment]
No, it will automatically close when the pull request is merged.

**Comment by tacaswell** [comment]
Closed by #6337 

@gepcel if you put `closes #ISSUENUMBER` in a git commit message that is what triggers the auto closing.
